### PR TITLE
Make messaging service options configurable programmatically and via configuration files

### PR DESCRIPTION
**Pull request by kuujo**
This PR exposes some internal Netty configurations to users via both configuration files and the Java API. The only Netty options that are currently exposed is the connection timeout. More importantly, the TLS configuration is exposed, so users can now configure and enable TLS.

### CODE REVIEW

**Comment by johnou** [comment]
I'll rebase https://github.com/atomix/atomix/pull/770 when this is merged.

**Comment by johnou** [comment]
Just one nit on the default config.

File: cluster/src/main/java/io/atomix/cluster/messaging/MessagingConfig.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+import io.atomix.utils.config.Config;
+
+import java.time.Duration;
+
+/**
+ * Messaging configuration.
+ */
+public class MessagingConfig implements Config {
+  private Duration connectTimeout = Duration.ofSeconds(1);
```
[the remark this answers is not in the document]
Comment (kuujo) [reply cluster/src/main/java/io/atomix/cluster/messaging/MessagingConfig.java]: I think that was a typo. Hahaha. Thanks!

**Comment by kuujo** [comment]
Need to cherry pick to the 3.1 branch

**Pull request merged**